### PR TITLE
Interaction Mode

### PR DESCRIPTION
**Pull request by TheChosenEvilOne**
## About The Pull Request
Re-adds intents alongside combat mode. I am opening this as a draft because this is (as of writing) a 348 file PR and I haven't finished converting all interactions to the new system.

## Why It's Good For The Game
I like intents.

## Changelog
:cl:
add: A changed version of intents have been re-added as a user preference.
/:cl:

### CODE REVIEW

**Comment by JinGiant** [comment]
Would intents and combat mode have to be maintained separately

**Comment by TheChosenEvilOne** [comment]
> Would intents and combat mode have to be maintained separately

no

**Comment by Kapu1178** [comment]
finish this so i can steal it thanks 👍

**Comment by TheChosenEvilOne** [comment]
I have decided to axe four intent mode in favour of three intent mode for technical reasons.

**Comment by dwasint** [comment]
How does this work with right clicks? did you just remap right click with intents to something else or?

**Comment by stanalbatross** [comment]
image
image
sprites are unaligned, literally unplayable

**Comment by TheChosenEvilOne** [comment]
> image image sprites aren't the same when on/off, literally unplayable

I made a mistake when copy pasting, it is funny though.

**Comment by NiftyRyanyan** [comment]
I really don't see the appeal for this if right clicking still is disarm, That was the main problem with CM. This is really just CM but with a extra grab button at it's current state.

**Comment by rdragan** [comment]
Cyborgs need to be fixed before we can merge this.

**Comment by TheChosenEvilOne** [comment]
> Cyborgs need to be fixed before we can merge this.

done

**Comment by SweptWasTaken** [comment]
"Refined" GitHub crashed when I tried to review it but my brief gaze upon it as well as checking it out in-game with no noticeable bugs fills me with determination.

Really great job TCEO, well done!

File: code/datums/interaction/interaction_mode.dm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/datum/interaction_mode/proc/reload_hud(mob/M)
+	if (UI)
+		owner.mob.hud_used.static_inventory -= UI
+	if (M.hud_used.has_interaction_ui)
+		M.hud_used.static_inventory += procure_hud(owner.mob, owner.mob.hud_used)
+
+/datum/interaction_mode/proc/replace(datum/interaction_mode/IM)
+	IM = new IM (owner)
+	if (UI)
+		UI?.hud.static_inventory -= UI
```
Comment (EdgeLordExe) [inline code/datums/interaction/interaction_mode.dm:33]: if U exists then ?. is just slower .

**Review by EdgeLordExe — APPROVED** [review]
LGTM

**Pull request merged**